openwrt-openwisp-monitoring/files/monitoring.agent

```diff
@@ -0,0 +1,67 @@
+#!/bin/sh
+
+while [ -n "$1" ]; do
+	case "$1" in
+		--url) export BASE_URL="$2"; shift;;
+		--uuid) export UUID="$2"; shift;;
+		--key) export KEY="$2"; shift;;
+		--verify_ssl) export verify_ssl="$2"; shift;;
+		--interval) export INTERVAL="$2"; shift;;
+		--monitored_interfaces) export MONITORED_INTERFACES=$2; shift;;
+		-*)
+			echo "Invalid option: $1"
+			exit 1
+		;;
+		*) break;
+	esac
+	shift;
+done
+
+if [ "$verify_ssl" = 0 ]; then
+    curl_command='curl -k'
+else
+    curl_command='curl'
+fi
+
+url="$BASE_URL/api/v1/monitoring/device/$UUID/?key=$KEY"
+
+collect_data(){
+	n=0
+	logger -s "Collecting NetJSoN Monitoring data" \
+	       -t monitoring \
+	       -p daemon.info
+
+	until [ "$n" -ge 5 ]
+	do
+		echo "$(/usr/sbin/netjson_monitoring "$MONITORED_INTERFACES")" && break
+		
+		if [ "$n" -ne 5 ]; then
+			logger -s "Collecting data failed! Retrying" \
+			       -t monitoring \
+			       -p daemon.err
+		fi
+		n=$((n+1))
+		sleep 5
+	done
```

Comment:
I added this loop to retry for 5 times(although it can be customisable). This will ensure errors like interrupted system calls don't create hurdles in the way

Reply:
log only if it fails more than the max amount of retries

Reply:
:+1: